yarn-project/simulator/src/public/fixtures/opcode_spammer.ts

```diff
@@ -0,0 +1,1081 @@
+    { offset: CALL_ARGS_SIZE, value: new Uint32(0) }, // argsSize = max uint32
+  ],
+  targetInstructions: () => [
+    new Call(0, CALL_L2_GAS_OFFSET, CALL_DA_GAS_OFFSET, CALL_ADDR_OFFSET, CALL_ARGS_SIZE, CALL_ARGS_OFFSET),
+  ],
+};
+
+/**
+ * Create bytecode that makes an external call in a loop.
+ *
+ * @returns the bytecode for the external call loop
+ */
+export function createExternalCallLoopBytecode(): Buffer {
+  const config = EXTERNAL_CALL_LOOP_CONFIG;
+  const instructions: Bufferable[] = [];
+
+  // 1. Setup memory
+  appendSetupInstructions(instructions, config.setup);
+  // 2. Infinite loop of external calls - maximize iterations until out-of-gas
+  appendInfiniteLoop(instructions, config);
+  return encodeToBytecode(instructions);
+}
+
+async function testStandardOpcodeSpam(
+  tester: PublicTxSimulationTester,
+  config: SpamConfig,
+  expectToBeTrue: (x: boolean) => void,
+): Promise<PublicTxResult> {
+  const bytecode = createOpcodeSpamBytecode(config);
+  const result = await deployAndExecuteCustomBytecode(bytecode, tester, config.label);
+
+  // should have halted with out of gas
+  expectToBeTrue(!result.revertCode.isOK());
+  const revertReason = result.findRevertReason()?.message.toLowerCase();
+  const allowedReasons = ['out of gas', 'not enough l2gas'];
+  // expect the reason to match ONE of the allowed reasons
+  expectToBeTrue(allowedReasons.some(allowedReason => revertReason!.includes(allowedReason)));
+  return result;
+}
+
+async function testSideEffectOpcodeSpam(
+  tester: PublicTxSimulationTester,
+  config: SpamConfig,
+  expectToBeTrue: (x: boolean) => void,
+): Promise<PublicTxResult> {
+  const innerBytecode = createSideEffectSpamBytecode(config);
+  const outerBytecode = createExternalCallLoopBytecode();
+  const innerContract = await deployCustomBytecode(innerBytecode, tester, `${config.label}_Inner`);
+  const outerContract = await deployCustomBytecode(outerBytecode, tester, `${config.label}_Outer`);
+  // Outer contract reads calldata[0] as inner contract address to CALL to
+  const result = await executeCustomBytecode(outerContract, tester, config.label, [innerContract.address.toField()]);
+
+  // should have halted with out of gas or explicit REVERT (assertion failed)
+  expectToBeTrue(!result.revertCode.isOK());
+  const revertReason = result.findRevertReason()?.message.toLowerCase();
+  const allowedReasons = ['assertion failed', 'out of gas', 'not enough l2gas'];
+  // expect the reason to match ONE of the allowed reasons
+  expectToBeTrue(allowedReasons.some(allowedReason => revertReason!.includes(allowedReason)));
+
+  // Top-level should _always_ run out of gas for these tests
+  // Check top-level halting message
+  // WARNING: only the C++ simulator (or TsVsCpp) will have haltingMessage
+  const outerCallMetadata = result.callStackMetadata[0] as CallStackMetadata;
+  expect(outerCallMetadata.haltingMessage?.toLowerCase()).toMatch(/out of gas|not enough l2gas/);
```

Comment:
I wonder if we should be checking the reasons, if this costs time due to callmetadata collection. Optionally we could leave the checks commented out (and not collect callmetadata). Again, I dont feel strongly.